core/src/main/java/io/grpc/transport/MessageFramer.java

```diff
@@ -166,13 +185,41 @@ private void writeFrame(InputStream message, int messageLength, boolean compress
     }
   }
 
+  /**
+   * Write a message that has been serialized to a sequence of buffers.
+   */
+  private void writeBufferChain(BufferChainOutputStream bufferChain, boolean compressed)
+      throws IOException {
+    ByteBuffer header = ByteBuffer.wrap(headerScratch);
```

Comment:
A safety net in case the allocator allocates something smaller than HEADER_LENGTH. That and having convenience methods for writing int/byte